Chapter4/Chapter4-1-1/YearMonth.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Chapter4_1_1 {
+    // 1.
+    /// <summary>
+    /// 年月クラス
+    /// </summary>
+    internal class YearMonth {
+        /// <summary>
+        /// 年
+        /// </summary>
+        public int Year { get; }
+        /// <summary>
+        /// 月
+        /// </summary>
+        public int Month { get; }
+
+        // 2.
+        /// <summary>
+        /// 21世紀
+        /// </summary>
+        public bool Is21Century => 2001 <= this.Year && this.Year <= 2100;
+
+        /// <summary>
+        /// YearMonthクラスのコンストラクタ
+        /// </summary>
+        /// <param name="vYear">年</param>
+        /// <param name="vMonth">月</param>
+        public YearMonth(int vYear, int vMonth) {
+            this.Year = vYear;
+            this.Month = vMonth;
+        }
+
+        // 3.
+        /// <summary>
+        /// 1カ月後を求めるメソッド
+        /// </summary>
+        /// <returns>1カ月後の年月</returns>
+        public YearMonth AddOneMonth() {
+            if (this.Month == 12) {
+                return new YearMonth(this.Year + 1, 1);
+            } 
+            return new YearMonth(this.Year, this.Month + 1);
+        }
```

Comment:
【FYI: 情報共有。アクション不要です。】
よいわるいは別にして参考まで。
あと、ここに出てくる12月とか1月ってマジックナンバーですよね。
適切な名前を付けられるといいですね。
```suggestion
        public YearMonth AddOneMonth() =>
            this.Month == 最後の月 ? new YearMonth(this.Year + 1, 最初の月) : new YearMonth(this.Year, this.Month + 1);
```

Reply:
コメントありがとうございます。勉強になります。
ちょうど昨日、マジックナンバーを排除する方が良いケースを例に、マジックナンバーについての知識を得ることができました。
今後は適切な名前を付けてコードを記述していきたいです。

Reply:
昔読んだ本に「1, 0, -1 以外はすべてマジックナンバーだ！」みたいな話がありました。
1, 0, -1 でもマジックナンバーのときはありますし、別に数字に限定した話でもないんですが、個人的には結構考え方の指針になってます。